http-client/src/main/java/io/micronaut/http/client/netty/NettyStreamedHttpResponse.java

```
@@ -140,7 +142,10 @@ public boolean isStream() {
     @Override
     public synchronized MutableHttpResponse<B> cookie(Cookie cookie) {
         ServerCookieEncoder.INSTANCE.encode(cookie)
-                .forEach(cookieEncoded -> headers.add(HttpHeaderNames.SET_COOKIE, cookieEncoded));
+                .forEach(cookieEncoded -> {
```

I think an exception would be better. Maybe with a configurable limit?

---

the limit of 4096 is enforced by browsers. Not sure what a configurable limit will do for users.

---

it apparently varies by browser: http://browsercookielimits.iain.guru/

and ofc there's non-browser clients

---

What are you thinking about? Making it configurable in [HttpServerConfiguration](https://github.com/micronaut-projects/micronaut-core/blob/4.10.x/http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java) and [HttpClientConfiguration](https://github.com/micronaut-projects/micronaut-core/blob/4.10.x/http-client-core/src/main/java/io/micronaut/http/client/HttpClientConfiguration.java)? That it is a bit hairy to do as HTTP responses classes are not beans, and to trickle down the configuration setting is a bit ugly. 

A simpler thing would be to add a default method to `MutableHttpResponse`  e.g. `MutableHttpResponse::cookie(Cookie cookie, Integer maxCookieSize)`, but I don't really like that API.

I personally think 4096 is a sensible default don't really see the use case. 

I have added [the size limit as a method parameter](https://github.com/micronaut-projects/micronaut-core/pull/11917/commits/848cd279067daab083ba3782ea2a5dabb33dcbbc) to make it more flexible.

---

> I think an exception would be better. Maybe with a configurable limit?

@yawkat I [have changed the implementation to thrown an exception instead of logging](https://github.com/micronaut-projects/micronaut-core/pull/11917/commits/6c233380eed2a248452186e45aee95ee927e4c96).